src/main/java/org/craftercms/studio/impl/v1/service/security/SecurityServiceImpl.java

```diff
@@ -125,19 +125,40 @@
     public String authenticate(@ValidateStringParam(name = "username") String username,
                                @ValidateStringParam(name = "password") String password) throws
         BadCredentialsException, AuthenticationSystemException, EntitlementException {
-        String toRet = securityProvider.authenticate(username, password);
-        if (StringUtils.isNotEmpty(toRet)) {
-            RequestContext requestContext = RequestContext.getCurrent();
-            HttpServletRequest httpServletRequest = requestContext.getRequest();
-            String ipAddress = httpServletRequest.getRemoteAddr();
 
+        String toRet = StringUtils.EMPTY;
+        RequestContext requestContext = RequestContext.getCurrent();
+        HttpServletRequest httpServletRequest = requestContext.getRequest();
+        String ipAddress = httpServletRequest.getRemoteAddr();
+        try {
+            securityProvider.authenticate(username, password);
+        } catch (BadCredentialsException | AuthenticationSystemException | EntitlementException e) {
+            ActivityService.ActivityType activityType = ActivityService.ActivityType.LOGIN_FAILED;
+            Map<String, String> extraInfo = new HashMap<String, String>();
+            extraInfo.put(DmConstants.KEY_CONTENT_TYPE, StudioConstants.CONTENT_TYPE_USER);
+            activityService.postActivity(getSystemSite(), username, ipAddress, activityType,
+                    ActivityService.ActivitySource.API, extraInfo);
+
+            logger.info("Failed to authenticate user " + username + " logging in from IP: " + ipAddress);
+
+            throw e;
+        }
+        if (StringUtils.isNotEmpty(toRet)) {
```

Comment:
`toRet` seems always to be empty, after declaring it's never assigned a value again